### PR TITLE
Fix budget slider responsiveness

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: https://github.com/woocommerce/woocommerce-android/issues/10574

### Description

These small changes fix several issues with the budget slider responsiveness. Issues are described in this PR: https://github.com/woocommerce/woocommerce-android/pull/10709#discussion_r1483263929

### Testing instructions

- Open Blaze campaign creation flow
- Open edit budget screen
- Play sliding back and forth with the budget slider. Ensure that the total budget amount is responsive, and it always displays a value that is multiple of the campaign's duration value. 

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/2663464/f43dd6ce-b56d-45e0-94d5-f9d0f9a3e805

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/budget/BlazeCampaignBudgetViewModel.kt [hunk 1]
```diff
@@ -192,7 +195,6 @@ class BlazeCampaignBudgetViewModel @Inject constructor(
     data class BudgetUiState(
         val currencyCode: String,
         val totalBudget: Float,
-        val sliderValue: Float,
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/budget/BlazeCampaignBudgetViewModel.kt:196]: The new approach is simpler and doesn't require to have 2 different values, one for the budget and other for the slider position

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by 0nko — APPROVED** [review]
The fix works great. Nice work! 🏅

**Pull request merged**